docs/documentation/siddhi-4.0.md

```diff
@@ -2220,51 +2223,61 @@ _We value your contribution on improving Siddhi and its extensions further._
 
 ### Writing Custom Extensions
 
-Custom extensions can be written in order to cater use case specific logic that are not available in Siddhi out of the box or as an existing extension. 
+Custom extensions can be written in order to cater use case specific logic that are not available in Siddhi out of the box or as an existing extension.
 
-There are five types of Siddhi extensions that you can write to cater your specific use cases. Please find each 
-extension types and related maven archetypes below. You can use these archetypes to generate maven projects for each 
+There are five types of Siddhi extensions that you can write to cater your specific use cases. These 
+extension types and the related maven archetypes are given below. You can use these archetypes to generate Maven projects for each 
 extension type.
 
-* Follow one of the step below, based on your project :
+* Follow the procedure for the required archetype, based on your project:
 
 
 **siddhi-execution**
 
-Siddhi-execution provides following extension types,
+Siddhi-execution provides following extension types:
 
 * Function
 * Aggregate Function
 * Stream Function
 * Stream Processor
 * Window
 
-You can use one or more from above mentioned extension types and implement according to your requirement. You can find more information about these extension types under the heading 'Extension types' in this document. 
+You can use one or more from above mentioned extension types and implement according to your requirement. 
+
+For more information about these extension types, see [Extension Types](#ExtensionTypes).
 
-* Run the following command
+To install and implement the siddhi-io extension archetype, follow the procedure below:
+
```

Comment:
Can we also add a note saying that function, aggregate function, window, stream processor and stream processor function will fall under execution type? Or will it be better to classify the extensions under the extension types itself?